twine/settings.py

```diff
@@ -136,7 +138,7 @@ def password(self) -> Optional[str]:
             # Workaround for https://github.com/python/mypy/issues/5858
             return cast(Optional[str], self.auth.password)
 
-    def _allow_noninteractive(self) -> ContextManager[None]:
+    def _allow_noninteractive(self) -> contextlib.AbstractContextManager[None]:
```

Comment:
This avoids an [error in the `docs` build](https://github.com/pypa/twine/runs/5077633699?check_suite_focus=true#step:5:66):

```
WARNING: py:obj reference target not found: typing.AbstractContextManager
```

By taking advantage of [PEP 585](https://www.python.org/dev/peps/pep-0585/) via `from __future__ import annotations` (noting that [`typing.ContextManager`](https://docs.python.org/3/library/typing.html#typing.ContextManager) is deprecated in Python 3.9). There's a lot more that could be done, e.g. replacing `Dict` with `dict` and `Optional[str]` with `str | None`, but I think that would be best handled in a separate PR.